static/js/components/Source.jsx

```diff
@@ -63,6 +70,63 @@ const Source = ({ route }) => {
     );
   }
 
+  const groupBy = (array, key) => array.reduce((result, cv) => {
+    (result[cv[key]] = result[cv[key]] || []).push(cv);
+    return result;
+  }, {});
+  function showClassification() {
+    // Here we compute the most recent non-zero probability class for each taxonomy
+
+    const filteredClasses = source.classifications.filter((i) => i.probability > 0);
+    const groupedClasses = groupBy(filteredClasses, 'taxonomy_id');
+    const sortedClasses = [];
+
+    Object.keys(groupedClasses).forEach((item) => sortedClasses.push(
+      groupedClasses[item].sort((a, b) => (a.modified < b.modified ? 1 : -1))
+    ));
+
+
+    if (sortedClasses.length > 0) {
+      return (
+        <div>
+          <b>Classification: </b>
+          {sortedClasses.map((c) => {
+            let name = taxonomyList.filter((i) => i.id === c[0].taxonomy_id);
+            if (name.length > 0) {
+              name = name[0].name;
+            }
+            return (
```

Comment:
what's with the return statement inside another return statement? not sure what's going on here or if this is even valid

Reply:
It's inside a map which is inside an if/then clause. Adding some comments inline.

Reply:
Ah right, I see now.